c10/util/numa.cpp

```diff
@@ -0,0 +1,132 @@
+#include "c10/util/numa.h"
+
+C10_DEFINE_bool(caffe2_cpu_numa_enabled, false, "Use NUMA whenever possible.");
+
+#if defined(__linux__) && !defined(C10_DISABLE_NUMA) && C10_MOBILE == 0
+#include <numa.h>
+#include <numaif.h>
+#define C10_NUMA_ENABLED
+#endif
+
+namespace c10 {
+
+#ifdef C10_NUMA_ENABLED
+bool IsNUMAEnabled() {
+  return FLAGS_caffe2_cpu_numa_enabled && numa_available() >= 0;
+}
+
+void NUMABind(int numa_node_id) {
+  if (numa_node_id < 0) {
+    return;
+  }
+  if (!IsNUMAEnabled()) {
+    VLOG(1) << "NUMA is not enabled";
+    return;
+  }
+
+  CAFFE_ENFORCE(
+      numa_node_id <= numa_max_node(),
+      "NUMA node id " + c10::to_string(numa_node_id) + " is unavailable");
+
+  auto bm = numa_allocate_nodemask();
+  numa_bitmask_clearall(bm);
```

Comment:
`numa_allocate_nodemask` is specified in the documentation to return a zero-filled bitmask. Thus, the clearall call is not necessary.

Reply:
cc @ilia-cher

Reply:
good catch